chpl/chpl-service/src/test/java/gov/healthit/chpl/manager/auth/UserManagerTest.java

```diff
@@ -0,0 +1,450 @@
+        UserManager userManager = new UserManager(null, userDAO, userResetTokenDAO, null, null, null, null);
+        UserResetTokenDTO token = userManager.createResetUserPasswordToken("user_a", "abc@def.com");
+
+        assertNotNull(token);
+    }
+
+    @Test(expected = UserRetrievalException.class)
+    public void createResetUserPasswordToken_UserIsNotValid_UserRetrievalExceptionThrown() throws UserRetrievalException {
+        Mockito.when(userDAO.findUserByNameAndEmail(ArgumentMatchers.anyString(), ArgumentMatchers.anyString()))
+                .thenReturn(null);
+
+        UserManager userManager = new UserManager(null, userDAO, userResetTokenDAO, null, null, null, null);
+        userManager.createResetUserPasswordToken("user_a", "abc@def.com");
+
+        fail();
+    }
+
+    @Test
+    public void authorizePasswordReset_ValidToken_ReturnTrue() {
+        UserManager userManager = new UserManager(env, null, userResetTokenDAO, null, null, null, null);
+        boolean auth = userManager.authorizePasswordReset("token");
+
+        assertEquals(true, auth);
+    }
+
+    @Test
+    public void authorizePasswordReset_TokenNotFound_ReturnFalse() {
+        Mockito.when(userResetTokenDAO.findByAuthToken(ArgumentMatchers.anyString()))
+                .thenReturn(null);
+
+        UserManager userManager = new UserManager(null, null, userResetTokenDAO, null, null, null, null);
+        boolean auth = userManager.authorizePasswordReset("token");
+
+        assertEquals(false, auth);
+    }
+
+    @Test
+    public void authorizePasswordReset_InvalidToken_ReturnFalse() {
+        Calendar oldDate = Calendar.getInstance();
+        oldDate.add(Calendar.HOUR, FIVE_HOURS_AGO);
+
+        Mockito.when(userResetTokenDAO.findByAuthToken(ArgumentMatchers.anyString()))
+                .thenReturn(UserResetTokenDTO.builder()
+                        .creationDate(oldDate.getTime())
+                        .build());
+
+        UserManager userManager = new UserManager(env, null, userResetTokenDAO, null, null, null, null);
+        boolean auth = userManager.authorizePasswordReset("token");
+
+        assertEquals(false, auth);
+    }
+
+    private UserDTO getUserDTO(Long id) {
+        return UserDTO.builder()
+                .accountEnabled(true)
+                .accountExpired(false)
+                .accountLocked(false)
+                .credentialsExpired(false)
+                .email("abc@def.com")
+                .failedLoginCount(0)
+                .friendlyName("Friendly Name")
+                .fullName("Full Name")
+                .id(id)
+                .lastLoggedInDate(new Date())
+                .phoneNumber("555-555-5555")
+                .signatureDate(new Date())
+                .subjectName("user_a")
+                .title("Sr Eng")
+                .build();
+    }
+
+    private MutableAcl getMutabeAcl() {
```

Comment:
I'm assuming this is an ACL service that we should only need for User Management tests, right? As all of our other things that used ACLs have been converted? I ask because if this is more generically useful then it would make sense to put it in that class with the Mockito / test helper things, right?